### PR TITLE
Making storage non-singleton

### PR DESCRIPTION
**Pull request by orsenkucher**
## Status
**READY**
On top of Synchronized PR

## Breaking Changes
NOT SURE, NO
Now, for example, clearing one storage instance won't clear other instance's cache, but will delete underlying files . But this is not a problem I think. 
Since `HydratedBlocDelegate` already has `final HydratedStorage storage` all Blocs will retrieve same instance.
All reads and writes are safe to perform from different instances, because `static final Lock`

## Description
Singleton makes things hard to test, so removing it.
Made `static final Lock _lock` private again. Since no more need in this workaround.

## Todos
- [x] Tests

## Impact to Remaining Code Base
This PR will affect:

* hydrated_bloc_storage.dart
* hydrated_bloc_test.dart

### CODE REVIEW

File: lib/src/hydrated_bloc_storage.dart [hunk 1]
```diff
@@ -22,7 +23,7 @@ abstract class HydratedStorage {
 /// Implementation of `HydratedStorage` which uses `PathProvider` and `dart.io`
 /// to persist and retrieve state changes from the local device.
 class HydratedBlocStorage implements HydratedStorage {
-  static HydratedBlocStorage _instance;
+  static final Lock _lock = Lock();
```
Comment (felangel) [inline lib/src/hydrated_bloc_storage.dart]:
```suggestion
  static final _lock = Lock();
```

File: test/hydrated_bloc_delegate_test.dart [hunk 2]
```diff
@@ -1,144 +0,0 @@
-import 'dart:convert';
```
Comment (felangel) [inline test/hydrated_bloc_delegate_test.dart:1]: same question here about why these tests were removed.

File: test/hydrated_bloc_storage_test.dart [hunk 3]
```diff
@@ -1,195 +0,0 @@
-import 'dart:convert';
```
Comment (felangel) [inline test/hydrated_bloc_storage_test.dart:1]: same question here about why these tests were removed.

File: test/hydrated_bloc_test.dart [hunk 4]
```diff
@@ -84,114 +103,485 @@ class MyMultiHydratedBloc extends HydratedBloc<int, int> {
...
+          );
+          final tasks = Iterable.generate(120, (i) => i).map((i) async {
+            final record = Iterable.generate(
+              i,
+              (i) => Iterable.generate(i, (j) => 'Point($i,$j);').toList(),
+            ).toList();
+            final jsoned = json.encode(record);
+            hydratedStorage.write(token, jsoned); // no await here
+
+            dynamic written;
```
Comment (felangel) [inline test/hydrated_bloc_test.dart]:
```suggestion
            Map<String, dynamic> written;
```

File: test/hydrated_bloc_test.dart [hunk 5]
```diff
@@ -84,114 +103,485 @@ class MyMultiHydratedBloc extends HydratedBloc<int, int> {
...
+            try {
+              hydratedStorage = await HydratedBlocStorage.getInstance(
+                storageDirectory: directory,
+              ); // basically refreshes cache
+              string = hydratedStorage.read(token) as String;
+              final object = json.decode(string);
+              written = object;
+            } on dynamic catch (_) {
+              written = null;
+              print(string);
```
Comment (felangel) [inline test/hydrated_bloc_test.dart]:
```suggestion
```

File: test/hydrated_bloc_test.dart [hunk 6]
```diff
@@ -84,114 +103,485 @@ class MyMultiHydratedBloc extends HydratedBloc<int, int> {
...
+            } // At least json is not corrupted
+            expect(written, isNotNull);
```
Comment (felangel) [inline test/hydrated_bloc_test.dart]: can we add an expect to assert that the value of `string` is what we expect?

File: test/hydrated_bloc_test.dart (hunk 4 above)
Comment (orsenkucher) [reply test/hydrated_bloc_test.dart]: `HydratedStorage` supports writing `dynamic` states, and here we write `List<List<String>>`

File: test/hydrated_bloc_test.dart (hunk 6 above)
Comment (orsenkucher) [reply test/hydrated_bloc_test.dart]: done👍

**Pull request closed without merging**